includes/discovery/discovery-protocols.inc.php

```diff
@@ -204,14 +216,42 @@
         $IndexId = key($lldp['lldpNeighborPortIndexId']);
 
         $local_ifName = $lldp['lldpNeighborPortId'][$IndexId];
-        $local_port_id = find_port_id('gigabitEthernet ' . $local_ifName, null, $device['device_id']);
 
+        //try to find our local port
+        $local_port_id = find_port_id('gigabitEthernet ' . $local_ifName, null, $device['device_id']); //normal port name
+        if (! $local_port_id) {
+            $local_port_id = find_port_id('gigabitEthernet ' . $local_ifName . ' : copper', null, $device['device_id']); // : copper
+        }
+        if (! $local_port_id) {
+            $local_port_id = find_port_id('gigabitEthernet ' . $local_ifName . ' : fiber', null, $device['device_id']); // : fiber
+        }
         $remote_device_id = find_device_id($lldp['lldpNeighborDeviceName'][$IndexId]);
         $remote_device_name = $lldp['lldpNeighborDeviceName'][$IndexId];
         $remote_device_sysDescr = $lldp['lldpNeighborDeviceDescr'][$IndexId];
         $remote_device_ip = $lldp['lldpNeighborManageIpAddr'][$IndexId];
-        $remote_port_descr = $lldp['lldpNeighborPortIdDescr'][$IndexId];
-        $remote_port_id = find_port_id($remote_port_descr, null, $remote_device_id);
+        $remote_device_mac = '';
+        if ($lldp['lldpNeighborChassisIdType'][$IndexId] == 'MAC address') {
+            $remote_device_mac = strtolower(str_replace(':', '', $lldp['lldpNeighborChassisId'][$IndexId]));
+        }
+
+        //need to use PortIdDescr first but it is broken on few older device
+        //if PortIdDescr is valid, then PortDescr is invalid and vice versa
+        if (strlen($lldp['lldpNeighborPortIdDescr'][$IndexId]) > 3) {
```

Comment:
Not sure your assumption that > 3 is valid (not sure it isn't either)

Reply:
Ok, explanation
T1600G-28TS 2.0 send these:
lldpNeighborPortIdDescr[49173][1] = "1"
lldpNeighborPortIdDescr[49174][1] = "1"
lldpNeighborPortIdDescr[49175][1] = "n"
lldpNeighborPortIdDescr[49175][2] = "4"

i saw one TP-LINK which have 2 random char
these are obviously wrong

proper data from devices: T2600G-18TS 2.0, T1600G-28TS 3.0,
lldpNeighborPortIdDescr[49167][1] = "eth1"
lldpNeighborPortIdDescr[49168][1] = "eth1"
lldpNeighborPortIdDescr[49170][1] = "gi1/0/5"

my assumption was that no portName will be shorter than 4 char
if you wish, i could lower down to >= 3

Reply:
You sure 1,1,4 are not ifIndex?

Reply:
> You sure 1,1,4 are not ifIndex?
even it IS a index, i see this only on 2 of 25 Jetstream
if you allow, i will leave this alone and use names

Reply:
> Not sure your assumption that > 3 is valid (not sure it isn't either)

lowered to >= 3